src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ConcurrencyLimiter.cs

```diff
@@ -319,5 +321,28 @@ protected override void Dispose(bool disposing)
 
             public CancellationTokenRegistration CancellationTokenRegistration { get; }
         }
+
+        private class WrappedTCS : TaskCompletionSource<RateLimitLease>
+        {
+            private readonly int PermitCount;
+            private readonly ConcurrencyLimiter Limiter;
+
+            public WrappedTCS(int permitCount, ConcurrencyLimiter limiter) : base(TaskCreationOptions.RunContinuationsAsynchronously)
+            {
+                PermitCount = permitCount;
+                Limiter = limiter;
+            }
+
+            public void TryCancel()
```

Comment:
Nit: there's a minor (internal) usability issue here, as now there's a functional difference between `TryCancel` and `TrySetCanceled`, which sound very similar.

Reply:
Would the recommended approach be to `new` override `TrySetCanceled`?